app/scripts/lib/transaction/smart-transactions.ts

```diff
@@ -169,12 +174,15 @@ class SmartTransactionHook {
       }
       const extensionReturnTxHashAsap =
         this.#featureFlags?.smartTransactions?.extensionReturnTxHashAsap;
-      this.#addApprovalRequest({
-        uuid,
-      });
-      this.#addListenerToUpdateStatusPage({
-        uuid,
-      });
+
+      if (shouldShowStatusPage) {
```

Comment:
Shouldn't this condition be also in the `#onApproveOrReject` function for
```
this.#controllerMessenger.call('ApprovalController:endFlow', {
  id: this.#approvalFlowId,
});
```

Reply:
Done in [03c74fd](https://github.com/MetaMask/metamask-extension/pull/28460/commits/03c74fd891bbda3ff3e0ad92e0f2aa00b644c19a)